cli/src/lib.rs

```diff
@@ -104,6 +109,9 @@ impl Cli {
                 }
             },
         };
+        if self.reduce {
+            expr.reduce(&json!({}));
```

Comment:
Rather than having to construct an empty JSON `Value`, this reads cleaner if we make the argument optional:

```suggestion
            expr.reduce(None);
```

Reply:
Yeah, definitely was planning on making it optional, this was a quick hack so that I could test some things with the CLI. The "partial reduce" (ie not necessarily boiling all the way down to a true or false) was something that clicked with me late in the day.